### DISCUSSION TITLE
MongoDbTestcontainerConfiguration Integration Tests get stuck when using mongo:latest version

### DISCUSSION THREAD

**Discussion by glendoci** [testcontainers/testcontainers-dotnet#636] [General]
Unable to use mongo:latest image version, my docker images do get created with the latest version but the integration tests hang none run and no data is added into them I have to manually stop the tests because they don't execute aka they don't even fail or succeed only stuck.

` _dbContainer = new TestcontainersBuilder<MongoDbTestcontainer>().WithDatabase(new MongoDbTestcontainerConfiguration { Database = "PrismPrivateApiGatewayDb", Username = "", Password = "" }).Build();
 await _dbContainer.StartAsync();`

I am using testcontainers to spin off temporary mongodb docker containers to run my integration tests using xunit so that I don't need to write any tear down methods. 

The above code will spin off a mongo:5.0.6 version instead of mongo:latest I tried adding a .WithImage("mongo:latest") but the StartAsync keeps hanging.

I tried new MongoDbTestcontainerConfiguration { Database = "PrismPrivateApiGatewayDb", Username = "", Password = "", Image= "mongo:latest" } but the property has a get; only and not a set...

**Comment by HofmeisterAn** [comment]
Can you please format your questions. It is unlikely that someone will even read it that way.

Regarding your issue. You can override the base image either with:

```csharp
_ = new MongoDbTestcontainerConfiguration("mongo:latest")
```

or

```csharp
.WithDatabase(new MongoDbTestcontainerConfiguration())
.WithImage("mongo:latest")
```

But the latest MongoDB Docker image (since version 6) has some breaking changes that are not compatible with the current configuration. As far as I know, it replaces `mongo` with `mongosh`, which we use here:

https://github.com/testcontainers/testcontainers-dotnet/blob/a5d45f88fbe78c2b831bd5b4c5bdc2ca5a1d56cc/src/Testcontainers/Configurations/Modules/Databases/MongoDbTestcontainerConfiguration.cs#L53-L54

I was trying to change the CLI executable to `mongosh`, but the container still does not start. There is probably a bit more to do, to support the version `>=6`.

**Reply by glendoci** [reply]
Hi, thanks for the reply I've tried both the proposed solutions above the other day which they do generate the docker containers with the image being the mongo:latest but the test freeze/left hanging they neither fail nor succeed and I went inside the containers to check if there was any data added but nothing tests don't even get to run and I am forced to manually stop them. They stayed like frozen for over 15min with nothing happening not even an error.

**Reply by HofmeisterAn** [reply]
As I said, `mongo:latest` aka version 6 is not compatible with the current default configuration (which relies on version 5). Testcontainers for .NET "hangs" with version 6, because the container never gets to the state that it is ready to use (completes the wait strategy). If you would like to use version 6, you need to figure out why the container does not start / gets to the ready state (one step is replacing `mongo` with `mongosh`). Sorry, right now, I cannot look into it in detail.

**Reply by glendoci** [reply]
No worries, couldn't get it to stop hanging will just use version 5 for now it's nothing major I can live with version 5 until support for version 6 for .net core is added there is no rush for me whenever you got time to implement it days weeks or months for me is the same.

I was just introducing new Integration Tests for the company I work for and found your library on a YouTube video and really loved the concept.

Thanks for getting back to me really appreciated and amazing concept and library, keep up the great work!
